include/seqan3/alignment/aligned_sequence/debug_stream_alignment.hpp

```diff
@@ -104,17 +104,30 @@ namespace seqan3
  *
  * \return The given stream to which the alignment representation is appended.
  */
+#if 0
 template <typename char_t, typename alignment_t>
     requires (detail::debug_streamable_tuple<alignment_t>
               && detail::all_model_aligned_seq<detail::tuple_type_list_t<std::remove_cvref_t<alignment_t>>>)
 inline debug_stream_type<char_t> & operator<<(debug_stream_type<char_t> & stream, alignment_t && alignment)
 {
+#else
+template <typename alignment_t>
+    requires (tuple_like<std::remove_cvref_t<alignment_t>> && detail::all_model_aligned_seq<detail::tuple_type_list_t<std::remove_cvref_t<alignment_t>>>)
+struct alignment_printer<alignment_t>
+{
+    constexpr static auto print = [](auto & stream, auto && alignment)
```

Comment:
Note that this printer definition does not need to know ANYTHING about `seqan3::debug_stream`.

Printer Definition is highly local and highly decoupled.

And furthermore, as long as `alignment_printer` is not explicitly instantiated, the compiler will just do a quick look if the entity names are defined. That means the risk of weird compiler behaviour, infinite recursion, stupid requires rules, etc.., is mostly circumvented, since the code is "dead"-code for the compiler, as long as it is not USED. This tremendously helps to reduce the risk during major compiler migrations.